cpp/src/parquet/column_writer.cc

```diff
@@ -949,11 +954,17 @@ void ColumnWriterImpl::BuildDataPageV2(int64_t definition_levels_rle_size,
   ResetPageStatistics();
 
   int32_t num_values = static_cast<int32_t>(num_buffered_values_);
-  int32_t null_count = static_cast<int32_t>(page_stats.null_count);
+  int32_t null_count = static_cast<int32_t>(num_buffered_nulls_);
   int32_t num_rows = static_cast<int32_t>(num_buffered_rows_);
   int32_t def_levels_byte_length = static_cast<int32_t>(definition_levels_rle_size);
   int32_t rep_levels_byte_length = static_cast<int32_t>(repetition_levels_rle_size);
 
+  // page_stats.null_count is not set when page_statistics_ is nullptr. It is only used
+  // here for safety check.
+  if (page_stats.has_null_count && page_stats.null_count != num_buffered_nulls_) {
```

Comment:
Should this maybe be a `DCHECK` since users shouldn't be able to influence these values and wouldn't understand the exception?

Reply:
I agree. Will switch to `DCHECK` accordingly.